Remotion/Validation/UnitTests/RuleBuilders/AddingObjectValidationRuleBuilderTest.cs

```diff
@@ -14,61 +14,153 @@
+
+      _addingObjectValidationBuilder.CanBeRemoved();
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule ()
     {
+      var metaValidationRuleStub = new Mock<IObjectMetaValidationRule>();
+
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(metaValidationRuleStub.Object))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(metaValidationRuleStub.Object);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_FuncOverload ()
     {
+      var fakeValidationResult = MetaValidationRuleValidationResult.CreateValidResult();
+
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(v => fakeValidationResult);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First(), Is.EqualTo(fakeValidationResult));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsTrue ()
     {
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule<IObjectValidator>(v => true);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().IsValid, Is.True);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsFalse ()
     {
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule<IObjectValidator>(v => false);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().IsValid, Is.False);
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().Message, Is.EqualTo(
+          "Meta validation rule 'v => False' failed for validator 'Remotion.Validation.Validators.IObjectValidator' on type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
```

Comment:
```suggestion
      var result = rule.Validate(new IObjectValidator[0]);
      Assert.That(result , Has.Exactly(1).Items);
      Assert.That(result , Has.All.With.Property(nameof(MetaValidationRuleValidationResult.IsValid)).False);
            Assert.That(result , Has.All.With.Property(nameof(MetaValidationRuleValidationResult.Message)).EqualTo(
          "Meta validation rule 'v => False' failed for validator 'Remotion.Validation.Validators.IObjectValidator' on type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
```
We could also combine the two conditions using `And` and then then we don't need the `result` variable. It's a matter of style and formatting.  I went with the above approach because it was easier to prepare in the review.